openwisp_monitoring/device/api/serializers.py

```diff
@@ -20,6 +25,72 @@ class Meta:
         model = DeviceMonitoring
 
 
+class WifiClientSerializer(FilterByOrganizationManaged, ValidatedModelSerializer):
+    class Meta:
+        fields = [
+            "mac_address",
+            "vendor",
+            "ht",
+            "vht",
+            "wmm",
+            "wds",
+            "wps",
+            "modified",
+            "created",
+        ]
+        model = WifiClient
+        read_only_fields = (
+            'created',
+            'modified',
+        )
+
+
+class WifiSessionCreateUpdateSerializer(
+    FilterByOrganizationManaged, ValidatedModelSerializer
+):
+    # When a relationship or ChoiceField has too many items,
+    # rendering the widget containing all the options can become very slow,
+    # and cause the browsable API rendering to perform poorly, Changed select field to input.
+    device = serializers.PrimaryKeyRelatedField(
+        queryset=Device.objects.all(), style={'base_template': 'input.html'}
+    )
+    wifi_client = serializers.PrimaryKeyRelatedField(
+        queryset=WifiClient.objects.all(), style={'base_template': 'input.html'}
+    )
```

Comment:
@Aryamanz29  can you double check if the same functionality can be added using `Meta.extra_kwargs`? I tried the following code and it seemed to be working. 

```python
class WifiSessionCreateUpdateSerializer(
    FilterByOrganizationManaged, ValidatedModelSerializer
):
    class Meta:
        model = WifiSession
        fields = ['device', 'wifi_client', 'ssid', 'interface_name']
        extra_kwargs = {
            'device': {'style': {'base_template': 'input.html'}},
            'wifi_client': {'style': {'base_template': 'input.html'}},
        }
```

Reply:
Yes, It works :)